src/MOIWrapper.jl

```diff
@@ -0,0 +1,373 @@
+    scaling2 = rev ? scaling / sqrt(2) : scaling * sqrt(2)
+    output = copy(coef)
+    diagidx = IntSet()
+    for i in 1:d
+        push!(diagidx, trimap(i, i))
+    end
+    for i in 1:length(output)
+        if rows[i] in diagidx
+            output[i] *= scaling
+        else
+            output[i] *= scaling2
+        end
+    end
+    output
+end
+function scalecoef(rows, coef, minus, s::MOI.PositiveSemidefiniteConeTriangle)
+    _scalecoef(rows, coef, minus, s.dimension, false)
+end
+function scalecoef(rows, coef, minus, ::Type{MOI.PositiveSemidefiniteConeTriangle})
+    _scalecoef(rows, coef, minus, sympackeddim(length(rows)), true)
+end
+_varmap(f) = map(vi -> vi.value, f.variables)
+_constant(s::MOI.EqualTo) = s.value
+_constant(s::MOI.GreaterThan) = s.lower
+_constant(s::MOI.LessThan) = s.upper
+constrrows(::MOI.AbstractScalarSet) = 1
+constrrows(s::MOI.AbstractVectorSet) = 1:MOI.dimension(s)
+constrrows(optimizer::SCSOptimizer, ci::CI{<:MOI.AbstractScalarFunction, <:MOI.AbstractScalarSet}) = 1
+constrrows(optimizer::SCSOptimizer, ci::CI{<:MOI.AbstractVectorFunction, <:MOI.AbstractVectorSet}) = 1:optimizer.cone.nrows[constroffset(optimizer, ci)]
+MOIU.canloadconstraint(::SCSOptimizer, ::Type{<:SF}, ::Type{<:SS}) = true
+MOIU.loadconstraint!(optimizer::SCSOptimizer, ci, f::MOI.SingleVariable, s) = MOIU.loadconstraint!(optimizer, ci, MOI.ScalarAffineFunction{Float64}(f), s)
+function MOIU.loadconstraint!(optimizer::SCSOptimizer, ci, f::MOI.ScalarAffineFunction, s::MOI.AbstractScalarSet)
+    a = sparsevec(_varmap(f), f.coefficients)
+    # sparsevec combines duplicates with + but does not remove zeros created so we call dropzeros!
+    dropzeros!(a)
+    offset = constroffset(optimizer, ci)
+    row = constrrows(s)
+    i = offset + row
+    # The SCS format is b - Ax ∈ cone
+    # so minus=false for b and minus=true for A
+    setconstant = _constant(s)
+    optimizer.cone.setconstant[offset] = setconstant
+    constant = f.constant - setconstant
+    optimizer.data.b[i] = scalecoef(row, constant, false, s)
+    append!(optimizer.data.I, fill(i, length(a.nzind)))
+    append!(optimizer.data.J, a.nzind)
+    append!(optimizer.data.V, scalecoef(row, a.nzval, true, s))
+end
+MOIU.loadconstraint!(optimizer::SCSOptimizer, ci, f::MOI.VectorOfVariables, s) = MOIU.loadconstraint!(optimizer, ci, MOI.VectorAffineFunction{Float64}(f), s)
+orderval(val, s) = val
+function orderval(val, s::MOI.PositiveSemidefiniteConeTriangle)
+    sympackedUtoL(val, s.dimension)
+end
+orderidx(idx, s) = idx
+function orderidx(idx, s::MOI.PositiveSemidefiniteConeTriangle)
+    sympackedUtoLidx(idx, s.dimension)
+end
+function MOIU.loadconstraint!(optimizer::SCSOptimizer, ci, f::MOI.VectorAffineFunction, s::MOI.AbstractVectorSet)
+    A = sparse(f.outputindex, _varmap(f), f.coefficients)
+    # sparse combines duplicates with + but does not remove zeros created so we call dropzeros!
+    dropzeros!(A)
+    colval = zeros(Int, length(A.rowval))
```

Comment:
Use `findnz` instead of manually constructing the list of column indices. It would be nicer to never touch the internal fields of `A`.